bin/propolis-cli/src/main.rs

```diff
@@ -187,11 +196,14 @@ async fn new_instance(
     memory: u64,
     disks: Vec<DiskRequest>,
     cloud_init_bytes: Option<String>,
+    silo_id: Uuid,
+    project_id: Uuid,
```

Comment:
I don't love seeing two arguments with the same type and different meanings, it seems quite easy to accidentally swap them and not notice --- would it make sense to use @sunshowers' `newtype_uuid` crate for this?

Reply:
Sure, that seems like an easy win!

Reply:
Done in 60e6c07